content/js1/sprints/4/prep/index.md

```diff
@@ -8,4 +8,666 @@ backlog= 'Module-JS1'
+
+In addition, we want our tests to:
+
+- be easy to write
+- be easy to read
+- to give clear feedback on what the current output is
+- to give clear feedback on what the target output is
+- easily write multiple test cases
+
+We can use a **test framework** to help us build test cases like this.
+
+{{<note type="definition" title="definition">}}
+A **test framework** is a set of tools we can use to build tests efficiently.
+{{</note>}}
+
+we can use a short dialogue to think about _why_ we want to use a testing framework:
+
+> "We need to implement a function"
+>
+> "Cool, how are you going to check it works?"
+>
+> "We can use tests to check that the function gives the target output as described in the specification"
+>
+> "No problem, how are you going to write a lot of tests efficiently?"
+>
+> "We can make use of a testing framework - this will help us to write test cases quickly. And it will ensure that the tests give us fast, reliable feedback."
+
+### 🎒 Starting a project
+
+Let's imagine we've started a brand new project in a directory called `week-4-example`.
+
+{{<note type="exercise" title="exercise">}}
+
+### 💻 Follow along
+
+Set up a new directory on your local machine called `week-4-example`.
+Make sure you're inside the `cyf` directory you created earlier in the course.
+Change directory into the this place and double check your current working directory.
+
+```bash
+$ pwd
+.../cyf/week-4-example
+```
+
+{{</note>}}
+
+To start, we need to create a file called `package.json` in our project.
+
+> A `package.json` is used to store useful information about our current project, like the name, description, version etc.
+
+We can create a `package.json` file and add the following to it:
+
+package.json
+
+```json
+{
+  "name": "week-4-test-example"
+}
+```
+
+Anyone using this project can now check the `package.json` to get the name of the project.
+We can also add a description:
+
+```json
+{
+  "name": "week-4-test-example",
+  "description": "An example application showing how to write tests using the jest framework"
+}
+```
+
+We can continue adding more information about our project as the project grows.
+For now we can double check we only have a `package.json` in our project:
+
+```bash
+$ ls
+package.json
+```
+
+### 📦 Using packages
+
+When writing software, we need to continually make use of **packages** written by other developers.
+
+A **package** is a set of programs which are grouped together to provide some functionality.
+
+Different programming languages give developers different ways of accessing packages for use in their code.
+
+In JavaScript, we can use **npm**.
+
+> **npm** is a tool used for accessing packages that can be downloaded and used to help developers write applications. The **npm registry** is an online collection of packages that we can download using **npm** commands.
+
+### 🃏 Installing Jest
+
+Our next step will be to figure out how to install the Jest package on our machine.
```

Comment:
```suggestion
Jest is a package used to help us to write and run test cases in JavaScript.

Our next step will be to figure out how to install the Jest package on our machine, so that we can use it in our project.
```